tests/cases/test_update.rs

```diff
@@ -0,0 +1,719 @@
+use common::*;
+use parameterized_macro::parameterized;
+use std::path::{Path, PathBuf};
+
+fn assert_missing_lockfile(lockfile: &PathBuf, name: &str) {
+    let lockfile = lal::Lockfile::from_path(&lockfile, name);
+    assert_eq!(
+        format!("{:>}", lockfile.err().unwrap()),
+        format!("No lockfile found for {}", &name)
+    );
+}
+
+fn assert_lockfile(lockfile: &PathBuf, name: &str, version: u32) {
+    let lockfile = lal::Lockfile::from_path(&lockfile, name).expect("read lockfile");
+    assert_eq!(lockfile.name, name.to_string());
+    assert_eq!(lockfile.version, version.to_string());
+}
+
+fn assert_manifest(component_dir: &Path, name: &str, version: u32) {
+    let manifest = lal::Manifest::read(&component_dir).expect("read manifest");
+    assert_eq!(
+        manifest.dependencies.get_key_value(name),
+        Some((&name.to_string(), &version))
+    );
+}
+
+fn assert_manifest_dev(component_dir: &Path, name: &str, version: u32) {
+    let manifest = lal::Manifest::read(&component_dir).expect("read manifest");
+    assert_eq!(
+        manifest.devDependencies.get_key_value(name),
```

Comment:
This was stabilized in 1.40, please upgrade your toolchains.

https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.get_key_value